### PR TITLE
Refactor CLI code

### PR DESCRIPTION
**Pull request by landrito**
This PR refactors the CLI code. This refactor is for a couple of different reasons:

1. Add a bunch of usage messages to the command line interface.
2. Make it such that adding commands to the command line interface is much easier to do because the usage message handling is handled by a cli manager.

### CODE REVIEW

File: cmd/daemon.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+		status, err := d.Install("start", "--port", InstallPort)
+		if err != nil {
+			ErrLog.Println(status, "\nError: ", err)
+			os.Exit(1)
+		}
+		StdLog.Println(status)
+	},
+}
+
+var daemonStartCmd = &cobra.Command{
```
Comment (pongad) [inline cmd/daemon.go]: I'm not very familiar with the daemon. What's the difference between installing and starting?

Comment (landrito) [reply cmd/daemon.go]: For linux:

`Install` creates a shell script that is placed at [/etc/init.d/](https://askubuntu.com/questions/5039/what-is-the-difference-between-etc-init-and-etc-init-d) I assume this is the common place to put daemon-like executables. 

`Start` calls the shell script to invoke the go program.

For other platforms, the commands do behaviorally similar things but with what is idiomatic for the platform.

**Comment by landrito** [comment]
This change also adds a StreamInterceptor which should resolve #10.

File: cmd/start.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+	if err == nil {
+		stdLog.Printf("    Returning Response: %+v\n", resp)
+	} else {
+		stdLog.Printf("    Returning Error: %+v\n", err)
+	}
+	stdLog.Println("")
+	return resp, err
+}
+
+type loggingServerStream struct {
```
[the remark this answers is not in the document]
Comment (landrito) [reply cmd/start.go]: Ah found the way.

File: cmd/daemon.go (hunk 1 above)
Comment (pongad) [reply cmd/daemon.go]: Ah

**Pull request merged**